[cant install] on mint20

ok previously I have tried compiling cmake and done many things none worked,

now I have installed linux mint 20 fresh, and Im trying to install neix

It has cmake 3.16.3

when doing

cmake .

I now get the error:

#################################ERROR BEGIN:
`-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is unknown
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- broken
CMake Error at /usr/share/cmake-3.16/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "/usr/bin/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/x/neix/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/make cmTC_4cad3/fast && /usr/bin/make -f CMakeFiles/cmTC_4cad3.dir/build.make CMakeFiles/cmTC_4cad3.dir/build
    make[1]: Entering directory '/home/x/neix/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_4cad3.dir/testCCompiler.c.o
    /usr/bin/cc    -o CMakeFiles/cmTC_4cad3.dir/testCCompiler.c.o   -c /home/x/neix/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_4cad3
    /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_4cad3.dir/link.txt --verbose=1
    /usr/bin/cc      CMakeFiles/cmTC_4cad3.dir/testCCompiler.c.o  -o cmTC_4cad3 
    /usr/bin/ld: cannot find Scrt1.o: No such file or directory
    /usr/bin/ld: cannot find crti.o: No such file or directory
    collect2: error: ld returned 1 exit status
    make[1]: *** [CMakeFiles/cmTC_4cad3.dir/build.make:87: cmTC_4cad3] Error 1
    make[1]: Leaving directory '/home/x/neix/CMakeFiles/CMakeTmp'
    make: *** [Makefile:121: cmTC_4cad3/fast] Error 2
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:2 (project)


CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/home/x/neix/CMakeFiles/CMakeOutput.log".
See also "/home/x/neix/CMakeFiles/CMakeError.log".
`


#################################ERROR END
Awaiting instructions.

This is what a beginner linux user probably will experience. And Im sort of a beginner at linux.

now it gives error at make

ERROR MESSAGE:

/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::Application()':
Application.cpp:(.text+0x5c): undefined reference to `initscr'
/usr/bin/ld: Application.cpp:(.text+0x61): undefined reference to `clear'
/usr/bin/ld: Application.cpp:(.text+0x66): undefined reference to `noecho'
/usr/bin/ld: Application.cpp:(.text+0x6b): undefined reference to `cbreak'
/usr/bin/ld: Application.cpp:(.text+0x75): undefined reference to `curs_set'
/usr/bin/ld: Application.cpp:(.text+0x99): undefined reference to `LINES'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::~Application()':
Application.cpp:(.text+0x155): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x15d): undefined reference to `wclrtoeol'
/usr/bin/ld: Application.cpp:(.text+0x162): undefined reference to `refresh'
/usr/bin/ld: Application.cpp:(.text+0x167): undefined reference to `endwin'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::createFeedWindow()':
Application.cpp:(.text+0x1f8): undefined reference to `COLS'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::createArticleWindow()':
Application.cpp:(.text+0x26e): undefined reference to `COLS'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::resize()':
Application.cpp:(.text+0x367): undefined reference to `endwin'
/usr/bin/ld: Application.cpp:(.text+0x36c): undefined reference to `refresh'
/usr/bin/ld: Application.cpp:(.text+0x371): undefined reference to `clear'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::printVersion()':
Application.cpp:(.text+0x3ef): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x401): undefined reference to `wattr_on'
/usr/bin/ld: Application.cpp:(.text+0x423): undefined reference to `mvprintw'
/usr/bin/ld: Application.cpp:(.text+0x42a): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x43c): undefined reference to `wattr_off'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::printControlHints()':
Application.cpp:(.text+0x457): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x469): undefined reference to `wattr_on'
/usr/bin/ld: Application.cpp:(.text+0x46f): undefined reference to `LINES'
/usr/bin/ld: Application.cpp:(.text+0x48a): undefined reference to `mvprintw'
/usr/bin/ld: Application.cpp:(.text+0x491): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x4a3): undefined reference to `wattr_off'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::show()':
Application.cpp:(.text+0x4dc): undefined reference to `refresh'
/usr/bin/ld: Application.cpp:(.text+0x4fb): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x503): undefined reference to `wgetch'
/usr/bin/ld: Application.cpp:(.text+0x976): undefined reference to `LINES'
/usr/bin/ld: Application.cpp:(.text+0x994): undefined reference to `mvprintw'
/usr/bin/ld: Application.cpp:(.text+0x999): undefined reference to `refresh'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_printWindow()':
ApplicationWindow.cpp:(.text+0xea): undefined reference to `box'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0xfa): undefined reference to `wrefresh'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_printPad()':
ApplicationWindow.cpp:(.text+0x114): undefined reference to `COLS'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x156): undefined reference to `prefresh'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_printContent()':
ApplicationWindow.cpp:(.text+0x18d): undefined reference to `wclear'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x22a): undefined reference to `wattr_on'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x25e): undefined reference to `mvwprintw'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x278): undefined reference to `wattr_off'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x2ae): undefined reference to `mvwprintw'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_create()':
ApplicationWindow.cpp:(.text+0x31d): undefined reference to `newwin'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x33a): undefined reference to `keypad'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x350): undefined reference to `newpad'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x36d): undefined reference to `keypad'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::hide()':
ApplicationWindow.cpp:(.text+0x405): undefined reference to `refresh'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x415): undefined reference to `wclear'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x425): undefined reference to `wclear'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::reset()':
ApplicationWindow.cpp:(.text+0x48b): undefined reference to `wclear'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::setPosition(int, int)':
ApplicationWindow.cpp:(.text+0x6b8): undefined reference to `mvwin'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x6de): undefined reference to `mvwin'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::setDimensions(int, int)':
ApplicationWindow.cpp:(.text+0x738): undefined reference to `wresize'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x757): undefined reference to `wresize'
/usr/bin/ld: bin/library/libneixFeed.a(FeedLoader.cpp.o): in function `neix::FeedLoader::loadXml()':
FeedLoader.cpp:(.text+0x315): undefined reference to `curl_global_init'
/usr/bin/ld: FeedLoader.cpp:(.text+0x31a): undefined reference to `curl_easy_init'
/usr/bin/ld: FeedLoader.cpp:(.text+0x35a): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x377): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x395): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x3b0): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x3cb): undefined reference to `curl_easy_setopt'
/usr/bin/ld: bin/library/libneixFeed.a(FeedLoader.cpp.o):FeedLoader.cpp:(.text+0x3e6): more undefined references to `curl_easy_setopt' follow
/usr/bin/ld: bin/library/libneixFeed.a(FeedLoader.cpp.o): in function `neix::FeedLoader::loadXml()':
FeedLoader.cpp:(.text+0x445): undefined reference to `curl_easy_perform'
/usr/bin/ld: FeedLoader.cpp:(.text+0x454): undefined reference to `curl_easy_cleanup'
/usr/bin/ld: FeedLoader.cpp:(.text+0x459): undefined reference to `curl_global_cleanup'
/usr/bin/ld: FeedLoader.cpp:(.text+0x47f): undefined reference to `curl_easy_strerror'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/neix.dir/build.make:91: bin/neix] Error 1
make[1]: *** [CMakeFiles/Makefile2:91: CMakeFiles/neix.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

unfortunately it did not work I think it is the same error:

[ 94%] Built target neixApplication
[ 94%] Built target neixHelper
[ 94%] Built target neixConfig
[ 94%] Built target neixParser
[ 94%] Built target neixFeed
[ 94%] Linking CXX executable bin/neix
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::Application()':
Application.cpp:(.text+0x5c): undefined reference to `initscr'
/usr/bin/ld: Application.cpp:(.text+0x61): undefined reference to `clear'
/usr/bin/ld: Application.cpp:(.text+0x66): undefined reference to `noecho'
/usr/bin/ld: Application.cpp:(.text+0x6b): undefined reference to `cbreak'
/usr/bin/ld: Application.cpp:(.text+0x75): undefined reference to `curs_set'
/usr/bin/ld: Application.cpp:(.text+0x99): undefined reference to `LINES'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::~Application()':
Application.cpp:(.text+0x155): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x15d): undefined reference to `wclrtoeol'
/usr/bin/ld: Application.cpp:(.text+0x162): undefined reference to `refresh'
/usr/bin/ld: Application.cpp:(.text+0x167): undefined reference to `endwin'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::createFeedWindow()':
Application.cpp:(.text+0x1f8): undefined reference to `COLS'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::createArticleWindow()':
Application.cpp:(.text+0x26e): undefined reference to `COLS'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::resize()':
Application.cpp:(.text+0x367): undefined reference to `endwin'
/usr/bin/ld: Application.cpp:(.text+0x36c): undefined reference to `refresh'
/usr/bin/ld: Application.cpp:(.text+0x371): undefined reference to `clear'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::printVersion()':
Application.cpp:(.text+0x3ef): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x401): undefined reference to `wattr_on'
/usr/bin/ld: Application.cpp:(.text+0x423): undefined reference to `mvprintw'
/usr/bin/ld: Application.cpp:(.text+0x42a): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x43c): undefined reference to `wattr_off'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::printControlHints()':
Application.cpp:(.text+0x457): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x469): undefined reference to `wattr_on'
/usr/bin/ld: Application.cpp:(.text+0x46f): undefined reference to `LINES'
/usr/bin/ld: Application.cpp:(.text+0x48a): undefined reference to `mvprintw'
/usr/bin/ld: Application.cpp:(.text+0x491): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x4a3): undefined reference to `wattr_off'
/usr/bin/ld: bin/library/libneixApplication.a(Application.cpp.o): in function `neix::Application::show()':
Application.cpp:(.text+0x4dc): undefined reference to `refresh'
/usr/bin/ld: Application.cpp:(.text+0x4fb): undefined reference to `stdscr'
/usr/bin/ld: Application.cpp:(.text+0x503): undefined reference to `wgetch'
/usr/bin/ld: Application.cpp:(.text+0x976): undefined reference to `LINES'
/usr/bin/ld: Application.cpp:(.text+0x994): undefined reference to `mvprintw'
/usr/bin/ld: Application.cpp:(.text+0x999): undefined reference to `refresh'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_printWindow()':
ApplicationWindow.cpp:(.text+0xea): undefined reference to `box'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0xfa): undefined reference to `wrefresh'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_printPad()':
ApplicationWindow.cpp:(.text+0x114): undefined reference to `COLS'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x156): undefined reference to `prefresh'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_printContent()':
ApplicationWindow.cpp:(.text+0x18d): undefined reference to `wclear'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x22a): undefined reference to `wattr_on'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x25e): undefined reference to `mvwprintw'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x278): undefined reference to `wattr_off'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x2ae): undefined reference to `mvwprintw'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::_create()':
ApplicationWindow.cpp:(.text+0x31d): undefined reference to `newwin'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x33a): undefined reference to `keypad'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x350): undefined reference to `newpad'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x36d): undefined reference to `keypad'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::hide()':
ApplicationWindow.cpp:(.text+0x405): undefined reference to `refresh'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x415): undefined reference to `wclear'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x425): undefined reference to `wclear'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::reset()':
ApplicationWindow.cpp:(.text+0x48b): undefined reference to `wclear'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::setPosition(int, int)':
ApplicationWindow.cpp:(.text+0x6b8): undefined reference to `mvwin'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x6de): undefined reference to `mvwin'
/usr/bin/ld: bin/library/libneixApplication.a(ApplicationWindow.cpp.o): in function `neix::ApplicationWindow::setDimensions(int, int)':
ApplicationWindow.cpp:(.text+0x738): undefined reference to `wresize'
/usr/bin/ld: ApplicationWindow.cpp:(.text+0x757): undefined reference to `wresize'
/usr/bin/ld: bin/library/libneixFeed.a(FeedLoader.cpp.o): in function `neix::FeedLoader::loadXml()':
FeedLoader.cpp:(.text+0x315): undefined reference to `curl_global_init'
/usr/bin/ld: FeedLoader.cpp:(.text+0x31a): undefined reference to `curl_easy_init'
/usr/bin/ld: FeedLoader.cpp:(.text+0x35a): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x377): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x395): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x3b0): undefined reference to `curl_easy_setopt'
/usr/bin/ld: FeedLoader.cpp:(.text+0x3cb): undefined reference to `curl_easy_setopt'
/usr/bin/ld: bin/library/libneixFeed.a(FeedLoader.cpp.o):FeedLoader.cpp:(.text+0x3e6): more undefined references to `curl_easy_setopt' follow
/usr/bin/ld: bin/library/libneixFeed.a(FeedLoader.cpp.o): in function `neix::FeedLoader::loadXml()':
FeedLoader.cpp:(.text+0x445): undefined reference to `curl_easy_perform'
/usr/bin/ld: FeedLoader.cpp:(.text+0x454): undefined reference to `curl_easy_cleanup'
/usr/bin/ld: FeedLoader.cpp:(.text+0x459): undefined reference to `curl_global_cleanup'
/usr/bin/ld: FeedLoader.cpp:(.text+0x47f): undefined reference to `curl_easy_strerror'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/neix.dir/build.make:91: bin/neix] Error 1
make[1]: *** [CMakeFiles/Makefile2:91: CMakeFiles/neix.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,8 +6,9 @@ find_package(CURL REQUIRED)
 include_directories(${CURL_INCLUDE_DIR})
 
 # Check if 'ncurses' is installed
+SET(CURSES_NEED_NCURSES TRUE)
 find_package(Curses REQUIRED)
-include_directories(${CURSES_INCLUDE_DIR})
+include_directories(${CURSES_INCLUDE_DIRS})
 
 # Prepare filesystem for installation
 file(MAKE_DIRECTORY $ENV{HOME}/.config/neix/)
@@ -60,7 +61,7 @@ add_library(neixConfig STATIC ${config})
 add_library(neixHelper STATIC ${helper})
 
 add_executable(neix ${main})
-target_link_libraries(neix neixApplication ${CURSES_LIBRARY})
+target_link_libraries(neix neixApplication ${CURSES_LIBRARIES})
 target_link_libraries(neix neixFeed ${CURL_LIBRARY})
 target_link_libraries(neix neixParser)
 target_link_libraries(neix neixConfig)

